### ISSUE TITLE
Failed to read configuration value at 'integrations.github[0].token' as it is not visible.

### ISSUE THREAD

**Issue by jcarres-mdsol** [backstage/backstage#11178]
I'm trying to make the ADR plugin work but getting `Failed to read configuration value at 'integrations.github[0].token' as it is not visible.`. 
I've read the documentation about visibility, the code for the configReader that outputs this message, I've experimented with a couple of things and still I have no idea how to make it work.

I think the very least there is a need to improve the documentation here.

**Comment by freben** [comment]
https://backstage.io/docs/conf/defining#visibility

https://github.com/backstage/backstage/blob/master/packages/integration/config.d.ts#L138

All tokens are actually marked as secret, and are therefore explicitly not available in the frontend at all, for security reasons.

**Comment by freben** [comment]
What you commonly want to do, is to make a small backend plugin "bridge" that exposes the specific functionality to want and handles the injection of the secret token. That plugin can then also for example have protection that only lets authenticated Backstage users access it (requiring a valid Backstage token to be presented in the auth header).

**Comment by jcarres-mdsol** [comment]
Currently we have oath(okta) integration so I guess only authenticated users can see anything in the system anyways. 

I guess the root problem here is that the ADR plugin requires the github token in the frontend, or that's what it seems that the error message is saying. This is very scary to me as that token would need to have permissions to read any repository, if would leak out it can be very dangerous. 

@kuangp  Am I missing something?

**Comment by freben** [comment]
No the secrets that are in config are actually completely cut out and never reach the frontend in the first place.

Alright so if this is happening in the new plugin, let's see if Phil can give some feedback on what could be happening.

**Comment by jcarres-mdsol** [comment]
I am not sure if it happens, I only have that error to go by, anyways, probably he has more context. It is an amazing idea, so looking forward!

**Comment by jcarres-mdsol** [comment]
I've checked other branch (no ADR plugin) and I still see this warning so maybe it is not related to the plugin at all. It happened that I upgraded and added the plugin at the same time so maybe some of the sofware/configuration changes I made created the warning and as I was not able to see the content by the plugin I was assuming it was that warning that stopped it from working

**Comment by kuangp** [comment]
I thought the ADR plugin hasn't been released yet 😅  are you working off a fork of the project? Like you mentioned, I'm pretty sure this is unrelated to the (frontend) ADR plugin since it uses the `ScmAuthApi` and from what I understand, it reads from your `auth` providers to request credentials from the user via Oauth instead of reading your PAT from `integrations`. 
Reference: https://github.com/backstage/backstage/blob/09500386511fcd71312266a9a1b93b0e38952b0c/plugins/adr/src/hooks/useOctokitRequest.ts#L31-L46

**Comment by jcarres-mdsol** [comment]
I'll dig deeper on this in a couple of days, I did have the annotation and nothing showed, probably I've done something wrong. 
The plugin was released as a nightly package. It is just such a great idea, I could not help but try it out ASAP 😄

**Comment by JordiPolo** [comment]
You are right, that's how that works and I've make it work. The warning in the title comes from somewhere else I am still not sure why.

Plugins do use the auth providers and thus an oauth application in github. I wonder @freben  is it even technically possible to have a github provider that works with the token similar to how the github integration works instead of needing to install a backstage oauth application?

**Comment by jcarres-mdsol** [comment]
Thanks for the response. Anyways this is really not an issue. Thanks for the responses!

**Comment by webark** [comment]
@freben I think we should reopen this.

the api `scmIntegrationsApiRef` is the culprit.

it's path is

https://github.com/backstage/backstage/blob/f0de00d4eb70a48cee8bd5c7031483a293c91478/packages/integration-react/src/api/ScmIntegrationsApi.ts#L29-L38
https://github.com/backstage/backstage/blob/f0de00d4eb70a48cee8bd5c7031483a293c91478/packages/integration/src/ScmIntegrations.ts#L57-L68
https://github.com/backstage/backstage/blob/f0de00d4eb70a48cee8bd5c7031483a293c91478/packages/integration/src/github/GithubIntegration.ts#L31-L33
https://github.com/backstage/backstage/blob/f0de00d4eb70a48cee8bd5c7031483a293c91478/packages/integration/src/github/config.ts#L166-L170
https://github.com/backstage/backstage/blob/f0de00d4eb70a48cee8bd5c7031483a293c91478/packages/integration/src/github/config.ts#L120-L126

So this will get called whenever you use that `scmIntegrationsApiRef` which is a front end api.

We should probably change to a FE only `readGithubIntegrationConfigs` correct? (Not sure if the other integrations in this list need an FE set also).
